### PR TITLE
GP2-377 - Track all pages

### PR DESCRIPTION
**Pull request by rafa-garcia**
- [x] Ticket exists in Jira  https://uktrade.atlassian.net/browse/GP2-377
 - [x] Jira ticket has the correct status.
 - [x] [Changelog](CHANGELOG.md) entry added.
 - [] This PR can be merged by reviewers. (If unticked, please leave for the author to merge)

### CODE REVIEW

File: core/mixins.py [hunk 1]
```diff
@@ -47,3 +49,14 @@ class AnonymousUserRequired:
 class AuthenticatedUserRequired:
     # used by core.wagtail_hooks.authenticated_user_required
     authenticated_user_required_redirect_url = constants.LOGIN_URL
+
+
+class WagtailGA360Mixin:
+    # We can't use GA360Mixin.get_context_data() because that was for a
+    # view not a model, so this is duplicated code :o(
```
Comment (stevejalim) [inline core/mixins.py:56]: Suggestion: I would turn this comment to a """docstring""" and mention what the inheriting model must set up in its `__init__`

File: core/models.py [hunk 2]
```diff
@@ -257,6 +262,15 @@ def get_template(self, request, *args, **kwargs):
 
     def get_context(self, request, *args, **kwargs):
         context = super().get_context(request)
+
+        self.set_ga360_payload(
+            page_id=self.id,
+            business_unit=settings.GA360_BUSINESS_UNIT,
+            site_section=self.slug,
```
Comment (stevejalim) [inline core/models.py:269]: Nice solution for what to put as the site_section - slugs are unique.

File: core/mixins.py [hunk 3]
```diff
@@ -47,3 +49,14 @@ class AnonymousUserRequired:
...
+    def remap_ga360_context_data_to_payload(self, request):
```
Comment (stevejalim) [inline core/mixins.py:57]: Minor suggestion `remap_ga360_context_data_to_payload` -> `add_ga360_data_to_payload` but it's a not an important thing :)

**Review by stevejalim — CHANGES_REQUESTED** [review]
Looking pretty cool! Thanks, dude!

Added a couple of places where I think some small changes/deletions are needed, and a few optional comments. 

Happy to re-review as soon as you've handled them - feel free to reject all the optional ones! 🎸

File: core/models.py [hunk 4]
```diff
@@ -210,6 +213,8 @@ class CMSGenericPage(
     mixins.EnableTourMixin,
     mixins.ExportPlanMixin,
     mixins.AuthenticatedUserRequired,
+    mixins.WagtailGA360Mixin,
+    GA360Mixin,
```
[the remark this answers is not in the document]
Comment (rafa-garcia) [reply core/models.py:217]: To have available the GA360Mixin method [set_ga360_payload()](https://github.com/uktrade/great-components/blob/8d0f13c8f89c1cff72a8d6fd78db0630f0005944/great_components/mixins.py#L67)

File: core/views.py [hunk 5]
```diff
@@ -62,13 +61,19 @@ def __init__(self):
...
             page_id='MagnaPage',
             business_unit='MagnaUnit',
-            site_section='MagnaSection',
-            site_subsection='MagnaSubsection',
+            site_section='login',
         )
     template_name = 'core/login.html'
 
 
 class SignupView(TemplateView):
```
[the remark this answers is not in the document]
Comment (rafa-garcia) [reply core/views.py:69]: I'll remove it from the one below, we don't yet need to capture subsection

[the remark this answers is not in the document]
Comment (stevejalim) [reply core/views.py:69]: But i thought this one needed `GA360Mixin` to be able to call `set_ga360_payload` ?

**Review by stevejalim — CHANGES_REQUESTED** [review]
Nearly there - i think we need the GA360Mixin used on one page (see comments) because the init of that object calls a method that the mixin gives us.

**Review by stevejalim — APPROVED** [review]
LGTM! 🏅 Hope the coverage stuff is OK to sort out! 👍

**Pull request merged**